src/EditorFeatures/Core/Implementation/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -107,8 +120,24 @@ private string GetRootPathWithGuid_NoLock()
                     var temporaryDocument = _workspace.CurrentSolution.AddProject(temporaryProjectInfoAndDocumentId.Item1)
                                                                      .GetDocument(temporaryProjectInfoAndDocumentId.Item2);
 
-                    var sourceFromMetadataService = temporaryDocument.Project.LanguageServices.GetService<IMetadataAsSourceService>();
-                    temporaryDocument = await sourceFromMetadataService.AddSourceToAsync(temporaryDocument, symbol, cancellationToken).ConfigureAwait(false);
+                    var useDecompiler = allowDecompilation;
+                    if (useDecompiler)
+                    {
+                        try
+                        {
+                            temporaryDocument = await DecompileSymbolAsync(temporaryDocument, _workspace.Options, symbol, cancellationToken).ConfigureAwait(false);
+                        }
+                        catch (Exception e) when (FatalError.ReportWithoutCrashUnlessCanceled(e))
+                        {
+                            useDecompiler = false;
```

Comment:
I would prefer to "see" the exception in debug builds instead of swallowing it, otherwise it might be hard to find out if there is a problem.